digdag-standards/src/main/java/io/digdag/standards/command/DockerCommandExecutor.java

```diff
@@ -32,40 +31,67 @@
 import static java.nio.charset.StandardCharsets.UTF_8;
 
 public class DockerCommandExecutor
-    implements CommandExecutor
+        implements CommandExecutor
 {
-    private final SimpleCommandExecutor simple;
-
     private static Logger logger = LoggerFactory.getLogger(DockerCommandExecutor.class);
 
+    private final CommandLogger clog;
+    private final SimpleCommandExecutor simple;
+
     @Inject
-    public DockerCommandExecutor(SimpleCommandExecutor simple)
+    public DockerCommandExecutor(final CommandLogger clog, final SimpleCommandExecutor simple)
     {
+        this.clog = clog;
         this.simple = simple;
     }
 
-    public Process start(Path projectPath, TaskRequest request, ProcessBuilder pb)
-        throws IOException
+    @Override
+    public CommandStatus run(final CommandContext context, final CommandRequest request)
+            throws IOException
     {
-        // TODO set TZ environment variable
-        Config config = request.getConfig();
+        final Config config = context.getTaskRequest().getConfig();
         if (config.has("docker")) {
-            return startWithDocker(projectPath, request, pb);
+            return runWithDocker(context, request);
         }
         else {
-            return simple.start(projectPath.toAbsolutePath(), request, pb);
+            return simple.run(context, request);
         }
     }
 
-    private Process startWithDocker(Path projectPath, TaskRequest request, ProcessBuilder pb)
+    private CommandStatus runWithDocker(final CommandContext context, final CommandRequest request)
+            throws IOException
     {
-        Config dockerConfig = request.getConfig().getNestedOrGetEmpty("docker");
+        // TODO set TZ environment variable
+        final Process p = startDockerProcess(context, request);
+
+        // copy stdout to System.out and logger
+        clog.copyStdout(p, System.out);
+
+        // Need waiting and blocking. Because the process is running on a single instance.
+        // The command task could not be taken by other digdag-servers on other instances.
+        try {
+            p.waitFor();
+        }
+        catch (InterruptedException e) {
+            throw Throwables.propagate(e);
+        }
+
+        return SimpleCommandStatus.of(request.getIoDirectory().toString(), p);
```

Comment:
`toString()` seems unnecessary.

Reply:
the first parameter type of `of` method is `String` type so, `toString` is necessary but, refactored https://github.com/treasure-data/digdag/pull/835/commits/ec1b5a8d25184974d24d115e41d1f1d90b689679